app/assets/javascripts/discourse/app/controllers/password-reset.js

```diff
@@ -136,6 +136,10 @@ export default Controller.extend(PasswordValidation, {
     },
 
     authenticateSecurityKey() {
+      this.set(
+        "selectedSecondFactorMethod",
+        SECOND_FACTOR_METHODS.SECURITY_KEY
+      );
```

Comment:
I'm tracing the code path and I don't think the root cause has been fixed yet. 

The problem seems to be here

https://github.com/discourse/discourse/blob/f66d317ee097dc584f459b878e3ae5a729621502/app/assets/javascripts/discourse/app/controllers/password-reset.js#L40

If I do a `console.log(this.model)` it is null which affects how `this.secondFactorMethod` is computed

https://github.com/discourse/discourse/blob/f66d317ee097dc584f459b878e3ae5a729621502/app/assets/javascripts/discourse/app/controllers/password-reset.js#L26-L30

Reply:
The model shouldn't be accessible inside `init()`, right? The problem I had with that line was that it wasn't reactive. So when it was used in the `SecurityKeyForm` component, there was a mismatch.

Reply:
Yes at first glance this doesn't sit right for me either, looking locally...

Reply:
> The model shouldn't be accessible inside `init()`, right?

Yup this is correct so whatever we are doing in `init()` is wrong and we need to fix that too.

Reply:
Its weird though, the reason why I put this in `init` yesterday and used `secondFactorMethod` computed property is because I couldn't access `this.model` directly, yet `this.secondFactorMethod` has the correct value in `init()`

Reply:
I followed up in https://github.com/discourse/discourse/pull/27363